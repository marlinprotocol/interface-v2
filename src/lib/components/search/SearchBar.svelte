<script lang="ts">
	import InputCard from '$lib/atoms/cards/InputCard.svelte';
	import { inputClasses } from '$lib/atoms/componentClasses';
	import { staticImages } from '$lib/components/images/staticImages';
	import { cn } from '$lib/utils/helpers/commonHelper';

	export let input: string;
	export let placeholder = '';
	export let disabled = false;
	export let styleClass = '';
	export let onSearchClick: (() => void) | undefined = undefined;
	export let label: string = '';
</script>

<InputCard
	variant="search"
	styleClass={cn('relative flex gap-4 items-center border border-grey-100', styleClass)}
>
	{#if label}
		<p
			class="absolute left-[18px] top-[-22%] mb-4 bg-white px-2 font-poppins text-sm font-light text-grey-800"
		>
			{label}
		</p>
	{/if}
	<img src={staticImages.searchV2Icon} alt="Search Icon" class="h-6 w-6" />
	<input
		prefix="0x"
		bind:value={input}
		on:click={(e) => onSearchClick?.()}
		id="address-display"
		class={cn(inputClasses.searchInputText, 'hideInputNumberAppearance')}
		{placeholder}
		{disabled}
	/>
</InputCard>
