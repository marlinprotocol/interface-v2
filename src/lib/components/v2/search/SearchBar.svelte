<script lang="ts">
	import search from 'svelte-awesome/icons/search';
	import Icon from '$lib/atoms/icons/Icon.svelte';
	import InputCard from '$lib/atoms/cards/InputCard.svelte';
	import { inputClasses } from '$lib/atoms/componentClasses';

	export let input: string;
	export let placeholder = '';
	export let disabled = false;
	export let styleClass = '';
	export let onSearchClick: (() => void) | undefined = undefined;
</script>

<InputCard variant="search" styleClass="{styleClass} flex gap-4 items-center">
	<Icon styleClass="opacity: .2" data={search} size={18} iconColorClass="icon-info" />
	<input
		prefix="0x"
		bind:value={input}
		on:click={(e) => onSearchClick?.()}
		id="address-display"
		class="hideInputNumberAppearance {inputClasses.searchInputText}"
		{placeholder}
		{disabled}
	/>
</InputCard>
