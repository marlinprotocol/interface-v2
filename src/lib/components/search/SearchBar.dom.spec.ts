import { describe, it, expect, vi } from 'vitest';
import { render, fireEvent } from '@testing-library/svelte';
import SearchBar from './SearchBar.svelte';

describe('SearchBar', () => {
	it("render's the component properly", () => {
		const { container } = render(SearchBar, {
			props: {
				placeholder: 'Search...',
				disabled: false,
				input: ''
			}
		});

		expect(container).toMatchSnapshot();
	});

	it('renders with correct placeholder and disabled state', () => {
		const { getByPlaceholderText } = render(SearchBar, {
			props: {
				placeholder: 'Search...',
				disabled: false,
				input: ''
			}
		});

		const inputElement = getByPlaceholderText('Search...') as HTMLInputElement;
		expect(inputElement).toBeTruthy();
		expect(inputElement.disabled).toBe(false);
	});

	it('calls onSearchClick when input is clicked', async () => {
		const mockOnSearchClick = vi.fn();
		const { getByPlaceholderText } = render(SearchBar, {
			props: {
				placeholder: 'Search...',
				onSearchClick: mockOnSearchClick,
				input: ''
			}
		});

		const inputElement = getByPlaceholderText('Search...') as HTMLInputElement;
		await fireEvent.click(inputElement);
		expect(mockOnSearchClick).toHaveBeenCalled();
	});

	it('binds input value correctly', async () => {
		const { getByPlaceholderText } = render(SearchBar, {
			props: {
				placeholder: 'Search...',
				input: 'Initial Value'
			}
		});

		const inputElement = getByPlaceholderText('Search...') as HTMLInputElement;
		expect(inputElement.value).toBe('Initial Value');

		await fireEvent.input(inputElement, { target: { value: 'New Value' } });
		expect(inputElement.value).toBe('New Value');
	});

	it('is disabled when disabled prop is true', () => {
		const { getByPlaceholderText } = render(SearchBar, {
			props: {
				placeholder: 'Search...',
				disabled: true,
				input: ''
			}
		});

		const inputElement = getByPlaceholderText('Search...') as HTMLInputElement;
		expect(inputElement.disabled).toBe(true);
	});
});
