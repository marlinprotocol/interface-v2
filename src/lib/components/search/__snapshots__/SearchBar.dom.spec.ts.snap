// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > render's the component properly 1`] = `
<body>
  <div>
    <div
      class="w-full bg-grey-200 px-4 py-2.5 rounded-[100px] relative flex gap-4 items-center border border-grey-100"
      data-testid="input-card"
    >
       
      <img
        alt="Search Icon"
        class="icon-invert h-6 w-6"
        src="/images/searchV2Icon.svg"
      />
       
      <input
        class="input input-ghost h-[30px] w-full text-base p-0 font-normal text-grey-800 focus:outline-none focus-within:border-b-2 focus-within:text-grey-800 focus:bg-transparent disabled:text-blue-500 placeholder-[#A8A8A8] hideInputNumberAppearance"
        id="address-display"
        placeholder="Search..."
        prefix="0x"
      />
    </div>
    <!--&lt;InputCard&gt;-->
    <!--&lt;SearchBar&gt;-->
  </div>
</body>
`;
